habitat/sims/habitat_simulator/habitat_simulator.py

```diff
@@ -192,74 +203,105 @@ class HabitatSim(habitat_sim.Simulator, Simulator):
         self._current_scene = self.sim_config.sim_cfg.scene_id
         super().__init__(self.sim_config)
+        # TODO Make action space MultiAgent compatible
         self._action_space = spaces.Discrete(
             len(self.sim_config.agents[0].action_space)
         )
         self._prev_sim_obs: Optional[Observations] = None
+        self._load_agent_body_meshes()
 
     def create_sim_config(
-        self, _sensor_suite: SensorSuite
+        self, _sensor_suite: Dict[int, SensorSuite]
     ) -> habitat_sim.Configuration:
         sim_config = habitat_sim.SimulatorConfiguration()
         # Check if Habitat-Sim is post Scene Config Update
         if not hasattr(sim_config, "scene_id"):
             raise RuntimeError(
                 "Incompatible version of Habitat-Sim detected, please upgrade habitat_sim"
             )
+
         overwrite_config(
             config_from=self.habitat_config.HABITAT_SIM_V0,
             config_to=sim_config,
         )
         sim_config.scene_id = self.habitat_config.SCENE
-        agent_config = habitat_sim.AgentConfiguration()
-        overwrite_config(
-            config_from=self._get_agent_config(), config_to=agent_config
-        )
-
-        sensor_specifications = []
-        for sensor in _sensor_suite.sensors.values():
-            sim_sensor_cfg = habitat_sim.SensorSpec()
+        agent_configs = []
+        for agent_id, sensor_suite in self.__sensor_suites.items():
+            agent_config = habitat_sim.AgentConfiguration()
             overwrite_config(
-                config_from=sensor.config, config_to=sim_sensor_cfg
+                config_from=self._get_agent_config(agent_id),
+                config_to=agent_config,
+                ignore_keys={"body_mesh_config"},
             )
-            sim_sensor_cfg.uuid = sensor.uuid
-            sim_sensor_cfg.resolution = list(
-                sensor.observation_space.shape[:2]
-            )
-            sim_sensor_cfg.parameters["hfov"] = str(sensor.config.HFOV)
-
-            # TODO(maksymets): Add configure method to Sensor API to avoid
-            # accessing child attributes through parent interface
-            # We know that the Sensor has to be one of these Sensors
-            sensor = cast(HabitatSimVizSensors, sensor)
-            sim_sensor_cfg.sensor_type = sensor.sim_sensor_type
-            sim_sensor_cfg.gpu2gpu_transfer = (
-                self.habitat_config.HABITAT_SIM_V0.GPU_GPU
+            sensor_specifications = []
+            for sensor in sensor_suite.sensors.values():
+                sim_sensor_cfg = habitat_sim.SensorSpec()
+                sim_sensor_cfg.sensor_subtype = getattr(
+                    habitat_sim.SensorSubType,
+                    sensor.config.SENSOR_SUBTYPE.upper(),
+                )
+                overwrite_config(
+                    config_from=sensor.config,
+                    config_to=sim_sensor_cfg,
+                    ignore_keys={
```

Comment:
I really want to have warnings about unsupported properties, but I am wondering if there is a way to keep this list form growing too much.

Reply:
all this keys aren't in HSIM spec?

Reply:
Nope or if they are they are converted in the loop below.